service/configurer/configurer.go

```diff
@@ -0,0 +1,109 @@
+package configurer
+
+import (
+	"github.com/spf13/afero"
+	"github.com/spf13/viper"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/draughtsman-operator/flag"
+	"github.com/giantswarm/draughtsman-operator/service/configurer/configmap"
+	"github.com/giantswarm/draughtsman-operator/service/configurer/file"
+	"github.com/giantswarm/draughtsman-operator/service/configurer/secret"
+	"github.com/giantswarm/draughtsman-operator/service/configurer/spec"
+)
+
+// Config represents the configuration used to create a Configurer.
+type Config struct {
+	// Dependencies.
+	FileSystem afero.Fs
+	K8sClient  kubernetes.Interface
+	Logger     micrologger.Logger
+
+	// Settings.
+	Flag  *flag.Flag
+	Viper *viper.Viper
+
+	Type spec.ConfigurerType
+}
+
+// DefaultConfig provides a default configuration to create a new Configurer
+// service by best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		FileSystem: nil,
+		K8sClient:  nil,
+		Logger:     nil,
+
+		// Settings.
+		Flag:  nil,
+		Viper: nil,
+	}
+}
+
+// New creates a new configured Configurer.
+func New(config Config) (spec.Configurer, error) {
+	// Settings.
+	if config.Flag == nil {
+		return nil, microerror.Maskf(invalidConfigError, "flag must not be empty")
+	}
+	if config.Viper == nil {
+		return nil, microerror.Maskf(invalidConfigError, "viper must not be empty")
+	}
+
+	var err error
+
+	var newConfigurer spec.Configurer
+	switch config.Type {
+	case configmap.ConfigurerType:
+		configmapConfig := configmap.DefaultConfig()
+
+		configmapConfig.K8sClient = config.K8sClient
+		configmapConfig.Logger = config.Logger
+
+		configmapConfig.Key = config.Viper.GetString(config.Flag.Service.Configurer.ConfigMap.Key)
+		configmapConfig.Name = config.Viper.GetString(config.Flag.Service.Configurer.ConfigMap.Name)
+		configmapConfig.Namespace = config.Viper.GetString(config.Flag.Service.Configurer.ConfigMap.Namespace)
+
+		newConfigurer, err = configmap.New(configmapConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+	case file.ConfigurerType:
+		fileConfig := file.DefaultConfig()
+
+		fileConfig.FileSystem = config.FileSystem
+		fileConfig.Logger = config.Logger
+
+		fileConfig.Path = config.Viper.GetString(config.Flag.Service.Configurer.File.Path)
+
+		newConfigurer, err = file.New(fileConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
```

Comment:
I tested locally using the file configurer. I do not expect the others to fail in production. When rolling this out we can test in lycan in the first place and then roll it out smoothly.